nativeruntime/cpp/CMakeLists.txt

```diff
@@ -1,50 +1,41 @@
 cmake_minimum_required(VERSION 3.10)
 
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+set(CMAKE_OSX_ARCHITECTURES "arm64;x86_64") # Universal libraries for Mac OS
+
 project(nativeruntime)
 
 # Some libutils headers require C++17
 set (CMAKE_CXX_STANDARD 17)
 
 find_package(JNI REQUIRED)
 
-# On Mac OS, search Homebrew for the icu4c distribution. The system version
-# does not include headers and static libraries.
-if (CMAKE_HOST_SYSTEM_NAME MATCHES "Darwin")
-  execute_process(
-    COMMAND brew --prefix icu4c
-    RESULT_VARIABLE BREW_ICU4C
-    OUTPUT_VARIABLE BREW_ICU4C_PREFIX
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-  )
+set(ANDROID_SQLITE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../external/sqlite")
 
-  if (NOT BREW_ICU4C EQUAL 0)
-    message( FATAL_ERROR "'brew --prefix icu4c' failed. Ensure homebrew is installed and run 'brew install icu4c'.")
-  endif()
-
-  if (BREW_ICU4C EQUAL 0 AND EXISTS "${BREW_ICU4C_PREFIX}")
-    message(STATUS "Found icu4c installed by Homebrew at ${BREW_ICU4C_PREFIX}")
-    list(APPEND CMAKE_PREFIX_PATH ${BREW_ICU4C_PREFIX})
-    find_library(BREW_ICUUC_LIBRARY libicuuc.a)
-    find_library(BREW_ICUI18N_LIBRARY libicui18n.a)
-    find_library(BREW_ICUDATA_LIBRARY libicudata.a)
-    include_directories(${BREW_ICU4C_PREFIX}/include)
-  endif()
-
-  if (NOT BREW_ICUUC_LIBRARY)
-    message(FATAL_ERROR "libicuuc.a not found. Please run 'brew install icu4c'")
-  endif()
+if(NOT EXISTS "${ANDROID_SQLITE_DIR}/dist/sqlite3.c")
+  message(FATAL_ERROR "SQLite submodule missing. Please run `git submodule update --init`.")
 endif()
 
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+set(ICU_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../external/icu")
 
-# Build flags derived from
-# https://cs.android.com/android/platform/superproject/+/android-11.0.0_r1:external/sqlite/dist/Android.bp
-set(ANDROID_SQLITE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../external/sqlite")
+if(NOT EXISTS "${ICU_DIR}/icu4c/source/i18n/ucol.cpp")
+  message(FATAL_ERROR "ICU submodule missing. Please run `git submodule update --init`.")
```

Comment:
Is it redundant with the checking in `configureICU`?

Reply:
It may be redundant, but I don't think it hurts to check these edge cases. For instance, if someone has a Robolectric checkout that already contains the sqlite submodule, and does a `git pull`, they will only have the sqlite submodule, but not the icu one. This actually happened to me. So I don't really think hurts to have the extra check.

Reply:
Thanks for explaining it. I don't catch this occasion when posting this question.